xdsl/__init__.py

```diff
@@ -1,3 +1,96 @@
+from collections.abc import Sequence
+from itertools import chain
+from types import ModuleType
+from typing import Any
+
+from xdsl.dialects import get_all_dialects
+from xdsl.traits import SymbolTable
+from xdsl.utils.dialect_stub import DialectStubGenerator
+
 from . import _version
 
 __version__ = _version.get_versions()["version"]
+
+import importlib.abc
+import importlib.util
+import os
+import sys
+
+
+class IRDLDialectLoader(importlib.abc.Loader):
+    """
+    Custom module loader for IRDL files.
+
+    When loading an irdl file as a module:
+    - parse the dialect
+    - generate its PyRDL implementation
+    - generate type stubs for its implmentation and write to disk
+    - populate the Python module
+    """
+
+    def __init__(self, module_name: str, path: str):
+        self.module_name = module_name
+        self.path = path
+
+    def exec_module(self, module: ModuleType):
+        from xdsl.context import MLContext
+        from xdsl.dialects.irdl import DialectOp
+        from xdsl.interpreters.irdl import make_dialect
+        from xdsl.parser import Parser
+
+        # Open the irdl file
+        with open(self.path) as file:
+            # Parse it
+            ctx = MLContext()
+            for dialect_name, dialect_factory in get_all_dialects().items():
+                ctx.register_dialect(dialect_name, dialect_factory)
+            irdl_module = Parser(ctx, file.read(), self.path).parse_module()
+
+            # Make it a PyRDL Dialect
+            dialect_name = os.path.basename(self.path)[:-5]
```

Comment:
why -5?

Reply:
It's "remove the .irdl extension"; would you rather have it be a library call or smth?

Reply:
I considered suggesting using Path but maybe it's overkill here. It may be worth adding `Drop ".irdl"` as a comment or using Path to remove the extension, your call.

Reply:
Done the call thing!